nucliadb/nucliadb/ingest/orm/knowledgebox.py

```diff
@@ -91,25 +91,14 @@ async def get_config(self) -> Optional[KnowledgeBoxConfig]:
             return self._config
 
     @classmethod
-    async def delete_kb(cls, txn: Transaction, slug: str = "", kbid: str = ""):
+    async def delete_kb(cls, txn: Transaction, kbid: str):
         # Mark storage to be deleted
         # Mark keys to be deleted
-        if not kbid and not slug:
-            raise AttributeError()
-
-        if slug and not kbid:
-            kbid_bytes = await txn.get(datamanagers.kb.KB_SLUGS.format(slug=slug))
-            if kbid_bytes is None:
-                raise datamanagers.exceptions.KnowledgeBoxNotFound()
-            kbid = kbid_bytes.decode()
-
-        if kbid and not slug:
-            kbconfig_bytes = await txn.get(datamanagers.kb.KB_UUID.format(kbid=kbid))
-            if kbconfig_bytes is None:
-                raise datamanagers.exceptions.KnowledgeBoxNotFound()
-            pbconfig = KnowledgeBoxConfig()
-            pbconfig.ParseFromString(kbconfig_bytes)
-            slug = pbconfig.slug
+        kb_config = await datamanagers.kb.get_config(txn, kbid=kbid)
+        if kb_config is None:
+            # consider KB as deleted
+            return
```

Comment:
Before we were raising an exception and now we don't. Is this OK? As long as all the callers of this method are adapted.

Reply:
The caller was the ingest gRPC endpoint and it wasn't adapted, it'd make the call fail with some internal error. We could manage the exception if you consider it a better approach

Reply:
I see backend has a big try-except for the delete, raising an unknown error if the gRPC call fails. Anyway, I think not raising an exception here is fine, we check KB existence using this maindb key